base/libgit2/types.jl

```diff
@@ -6,12 +6,20 @@ const OID_RAWSZ = 20
 const OID_HEXSZ = OID_RAWSZ * 2
 const OID_MINPREFIXLEN = 4
 
-immutable GitHash
+abstract AbstractGitHash
+
+immutable GitHash <: AbstractGitHash
     val::NTuple{OID_RAWSZ, UInt8}
     GitHash(val::NTuple{OID_RAWSZ, UInt8}) = new(val)
 end
 GitHash() = GitHash(ntuple(i->zero(UInt8), OID_RAWSZ))
 
+immutable GitShortHash <: AbstractGitHash
+    hash::GitHash
+    len::Csize_t
```

Comment:
should be extra careful (and add lots of comments) that this gets used consistently as the right notion of length everywhere, hex or raw